enhancements/image-references.md

```diff
@@ -0,0 +1,241 @@
+  containerized workloads on OpenShift
+- it is as simple for a developer to build a bundle referring to test versions
+  of operand images as it is for a CI or productized build pipeline to create
+  bundle images for publication
+- operator manifest authors dictate the set of image references in manifests
+  that must be resolved and pinned
+- operator bundle images may be inspected to determine the pull specifications
+  that were used in the creation of the bundle
+- operator manifest authors must not be required to define the registry from
+  which any individual build system will resolve image references
+- upstream operator manifests must not be required to know how common names or
+  references change when built in a downstream pipeline
+
+### Non-Goals
+
+- no prescriptive statement is made about the specific format or contents of the
+  bundle image layers; any will be transparently supported as an output
+
+## Proposal
+
+### User Stories
+
+#### Story 1
+
+As an author of a manifest, I would like to check in manifests to my upstream
+repository that are self-consistent, valid and make no assumptions about the
+build system that will eventually create a bundle image with them.
+
+#### Story 2
+
+As an author of an operand, I would like to create a bundle locally in order
+to test my operator end-to-end on a cluster of my choosing without having to
+edit the core configuration for the operator.
+
+#### Story 3
+
+As an author of a build system, I would like to operate with tooling that allows
+me to clearly define the source of truth for image digests in order to keep the
+build-system-specific configuration to an absolute minimum.
+
+#### Story 4
+
+As an engineer involved in publishing an optional operator, I would like to
+configure semantically equivalent image pull specifications once, in order to not
+need to configure each build system independently.
+
+### Implementation Details/Notes/Constraints
+
+The core problems that must be solved in the implementation of this proposal have
+already been handled in the workflow used in `oc adm release new`. When implementing
+improvements to the `operator-sdk generate bundle` process we will simply need to
+create a shared library for the two tools to use. While the shape of the output
+is slightly different and some of the semantics about how the output should be 
+formatted are dissmilar, the core image reference rewriting is identical and the
+process of building a `FROM scratch` image layer is also identical.
+
+Today, some prior art exists in the OSBS, IIB and CVP workflows for building
+operator bundles. As we improve the Operator SDK tooling to create a straghtforward
+process for creating bundle images, we must make sure a seamless migration is possible.
+
+### Risks and Mitigations
+
+It will be critical that the design be vetted by all of the concerned parties, from
+operator manifest authors to CI system authors and productized pipeline authors to
```

Comment:
AIUI there is a key difference between what is proposed here and how IIB currently works, which is that this method finds all references in the manifests directory and replaces them all. The existing IIB process only replaces in specific locations. I definitely prefer the approach in this enhancement as it gives more flexibility to operator authors and behaves the same way as openshift's image replacement used for its core payload building (giving us consistency in developer experience). However as far as some pipelines are concerned, it does mean it could potentially be replacing strings that were not being replaced before. Have we reached out to any operator authors yet to gauge whether this is going to be a concern?